src/verification/utils/definitions/disableNl.gobra

```diff
@@ -0,0 +1,137 @@
+// Copyright 2022 ETH Zurich
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//   http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+// +gobra
+
+package definitions
+
+// lemmas for working with disableNL
+
+ghost
+requires a == b
+ensures a * c == b * c
+ensures c * a == c * b
+decreases
+func LemmaMulReplace(a, b, c int) { }
+
+ghost
+requires a == b
+requires c == d * a
+ensures  c == d * b
+decreases
+func LemmaMulSubst1(a, b, c, d int) { }
+
+ghost
+requires a * 2 == a2
+requires b * 2 == b2
+requires a == c * b
+ensures a2 == c * b2
+decreases
+func LemmaMul2Inj(a, b, a2, b2, c int) {
+	assert a * 2 == (c * b) * 2
+	assert a * 2 == c * (b * 2)
+	assert a2 == c * (b * 2)
+	LemmaMulReplace(b*2, b2, c)
+	assert a2 == c * b2
+}
+
+// a <= b ==> a * c <= b * c
+ghost
+requires a <= b
+requires 0 <= c
+ensures a * c <= b * c
```

Comment:
```suggestion
ensures  a * c <= b * c
```

Reply:
please align the pres with the posts

Reply:
doing so in the .go files looks odd to me. I think when pre/post are separated by a blank line, not aligning them looks nicer. This is also in tune with gofmt, which does not align types in structs if the fields are separated by a blank line.

Note that I separated the pre/post conditions with a blank line for nicer hovering with gopls.